cloud/linode/loadbalancers.go

```diff
@@ -311,6 +305,35 @@ func (l *loadbalancers) updateNodeBalancer(
 				break
 			}
 		}
+		oldNBNodeIDs := make(map[string]int)
+		if currentNBCfg != nil {
+			// Obtain list of current NB nodes and convert it to map of node IDs
+			currentNBNodes, err := l.client.ListNodeBalancerNodes(ctx, nb.ID, currentNBCfg.ID, nil)
+			if err != nil {
+				// This error can be ignored, because if we fail to get nodes we can anyway rebuild the config from scratch,
+				// it would just cause the NB to reload config even if the node list did not change, so we prefer to send IDs when it is posible.
+				klog.Warningf("Unable to list existing nodebalancer nodes for NB %d config %d, error: %s", nb.ID, newNBCfg.ID, err)
+			} else {
+				for _, node := range currentNBNodes {
+					oldNBNodeIDs[node.Address] = node.ID
+				}
+			}
```

Comment:
```suggestion
			}
			for _, node := range currentNBNodes {
				oldNBNodeIDs[node.Address] = node.ID
			}
```

It's fine to range over a nil slice (nothing happens) so you can avoid the else case and the additional intendentation